lib/views/post/widgets/vote_button.dart

```diff
@@ -0,0 +1,170 @@
+import 'package:flutter/material.dart';
+import 'package:pr0gramm/constants/vote_constants.dart';
+import 'package:pr0gramm/entities/enums/vote.dart';
+import 'package:pr0gramm/entities/enums/vote_button_type.dart';
+import 'package:pr0gramm/services/vote_animation_service.dart';
+
+class VoteButton extends StatefulWidget {
+  final VoteAnimationService animationService;
+  final VoteButtonType type;
+  final bool disabled;
+
+  VoteButton({
+    Key key,
+    @required this.type,
+    this.disabled = false,
+    this.animationService,
+  }) : super(key: key);
+
+  @override
+  _VoteButtonState createState() {
+    return _VoteButtonState();
+  }
+}
+
+class _VoteButtonState extends State<VoteButton> with TickerProviderStateMixin {
+  AnimationController rotationController;
+  AnimationController colorController;
+  AnimationController favoriteController;
+  Animation<Color> colorTween;
+  Animation<double> favoriteMorph;
+  CurvedAnimation favoriteMorphCurveAnimation;
+  Color color;
+  Color lastColor;
+  IconData icon;
+
+  onStateChange(VoteAnimation voteAnimation) {
+    setState(() {
+      lastColor = color;
+      color = _getColorByAnimation(voteAnimation);
+
+      if (voteAnimation == VoteAnimation.voteFocused ||
```

Comment:
This is a very nested if. Use braces everywhere to improve readability

Reply:
will be obsolete when splitting into multiple Widgets